pre-commit

```diff
@@ -50,7 +50,8 @@ fi
 
 # Only puppet 3.2.1 - 3.8 support "--parser future" option.
 case $(puppet --version) in
-  4*) USE_PUPPET_FUTURE_PARSER="disabled" ;;
+  3.[2-8]*) USE_PUPPET_FUTURE_PARSER="enabled" ;;
```

Comment:
Instead of `3.[2-8]*` you should use `3\.(1\.[1-9]|[2-9]).*`

The "--parser=future" command-line setting was supported as early as version 3.1.1 (See [lib/puppet/version.rb](https://github.com/puppetlabs/puppet/blob/c3df220fca296307ad2d17bc8abb17e24b99f30b/lib/puppet/version.rb#L10) and [lib/puppet/defaults.rb](https://github.com/puppetlabs/puppet/blob/c3df220fca296307ad2d17bc8abb17e24b99f30b/lib/puppet/defaults.rb#L1511)).

If you want to be pedantic, you could use `3\.(1\.[1-9]|[2-7]|8\.[0-7]).*` since there were no versions between `3.8.7` and `4.0.0` but I think `3\.(1\.[1-9]|[2-9]).*` is simpler and clearer.